lib/HLSL/DxilCondenseResources.cpp

```diff
@@ -1630,10 +1636,13 @@ void UpdateStructTypeForLegacyLayout(DxilResourceBase &Res,
   Constant *Symbol = Res.GetGlobalSymbol();
   Type *ElemTy = Symbol->getType()->getPointerElementType();
   bool IsResourceArray = Res.GetRangeSize() != 1;
+  std::vector<unsigned> arrayDims;
   if (IsResourceArray) {
-    // Support Array of struct buffer.
-    if (ElemTy->isArrayTy())
+    // Support Array of ConstantBuffer.
+    while (ElemTy->isArrayTy()) {
```

Comment:
We should really have a place for snippets like this that we code again and again all the time:
```
Type* StripArrayTypes(Type *Ty, SmallVectorImpl<unsigned> *OuterToInnerLengths = nullptr);
Type* WrapInArrayTypes(Type *Ty, ArrayRef<unsigned> OuterToInnerLengths);
```

Reply:
Thanks, I implemented this as part of an update.  Also, tested some other array cases and found more bugs and fxc quirkiness to fix/match.